cmd/juju/charmhub/download.go

```diff
@@ -102,6 +105,8 @@ func (c *downloadCommand) SetFlags(f *gnuflag.FlagSet) {
 	f.IntVar(&c.revision, "revision", -1, "specify a revision of the charm to download")
 	f.StringVar(&c.archivePath, "filepath", "", "filepath location of the charm to download to")
 	f.BoolVar(&c.noProgress, "no-progress", false, "disable the progress bar")
+	f.BoolVar(&c.resources, "resources", false, "download the resources associated with the charm (will be DEPRECATED and default behaviour in 4.0)")
+	f.BoolVar(&c.noResources, "no-resources", false, "disable downloading the resources associated with the charm")
```

Comment:
suggestion: wait for this until downloading the resources are the default. It works, but a bit odd when both flags are specified. Does nothing if `--resources` is specified.

```
 juju download juju-qa-test  --resources --channel 2.0/edge --no-resources
Fetching charm "juju-qa-test" revision 23 using "2.0/edge" channel and base "amd64/ubuntu/22.04"
Install the "juju-qa-test" charm with:
    juju deploy ./juju-qa-test_r23.charm
```

Reply:
I can make it an error to pass both. The reason I added the `--no-resources` argument was to allow us to default to downloading the resources in 4.0, and have some compatibility overlap with 3.6. Happy to just make the hard change in 4.0 anyway.